peps/pep-0763.rst

```diff
@@ -0,0 +1,275 @@
+
+On a technical level, the problem of deletion is mitigated by
+"yanking," also specified in :pep:`592`. However, deletions continue to be
+allowed on PyPI, and have caused multiple notable disruptions to the Python
+ecosystem over the interceding years:
+
+* July 2022: `atomicwrites <https://pypi.org/project/atomicwrites/>`_
+  was `deleted by its maintainer <https://github.com/untitaker/python-atomicwrites/issues/61>`_
+  in an attempt to remove the project's "critical" designation, without the
+  maintainer realizing that project deletion would also delete all previously
+  uploaded versions.
+
+  The project was subsequently restored with the maintainer's consent,
+  but at the cost of manual administrator action and extensive downstream
+  breakage to projects like `pytest <https://github.com/pytest-dev/pytest/issues/10114>`_.
+  As of October 2024, atomicwrites is archived but still has
+  around `4.5 million monthly downloads from PyPI <https://pypistats.org/packages/atomicwrites>`_.
+
+* April 2023: `codecov <https://pypi.org/project/codecov/>`_ was deleted by
+  its maintainers after a long deprecation period. This caused extensive
+  breakage for many of Codecov's CI/CD users, who were unaware of the
+  deprecation period due to limited observability of deprecation warnings
+  within CI/CD logs.
+
+  The project was
+  `subsequently re-created <https://about.codecov.io/blog/message-regarding-the-pypi-package/>`_
+  by its maintainers, with a new release published to compensate for the deleted releases
+  (which were not restored), meaning that any pinned installations remained
+  broken. As of October 2024, this single release remains the only release on
+  PyPI and has around
+  `1.5 million monthly downloads <https://pypistats.org/packages/codecov>`_.
+
+* June 2023: `python-sonarqube-api <https://pypi.org/project/python-sonarqube-api/>`_
+  deleted all released versions prior to 2.0.2, which was both a breaking
+  change version *and* was accompanied by a paid "Professional Edition".
+
+  The project's maintainer subsequently
+  `deleted conversations <https://discuss.python.org/t/stop-allowing-deleting-things-from-pypi/17227/114>`_
+  and force-pushed over the tag history for ``python-sonarqube-api``'s source repository,
+  impeding efforts by the community to compare changes between versions
+  and obtain previous, permissively-licensed versions.
+
+* June 2024: `PySimpleGUI <https://pypi.org/project/PySimpleGUI/>`_ changed
+  licenses from LGPL to a commercial license, and deleted
+  `nearly all previous versions <https://discuss.python.org/t/48790/27>`_.
+  This resulted in widespread disruption for users, who (prior
+  to the relicensing) were downloading PySimpleGui
+  approximately 25,000 times a day. This deletion also impeded efforts
+  by the community to provide "continuity" projects, as the
+  permissively-licensed artifacts were no longer available for reference
+  and re-uploading.
```

Comment:
The suggestion is for consistent capitalisation, but as above, perhaps there's a way we can re-word this -- no one has an obligation to provide *free* software, so changing to a commercial licence isn't inherently a bad thing, which a couple of these examples seem to imply. Again perhaps it would be better simply to focus on the disruption caused downstream, without warning, &c &c.

```suggestion
* June 2024: `PySimpleGUI <https://pypi.org/project/PySimpleGUI/>`_ changed
  licenses from LGPL to a commercial license, and deleted
  `nearly all previous versions <https://discuss.python.org/t/48790/27>`_.
  This resulted in widespread disruption for users, who (prior
  to the relicensing) were downloading PySimpleGUI
  approximately 25,000 times a day. This deletion also impeded efforts
  by the community to provide "continuity" projects, as the
  permissively-licensed artifacts were no longer available for reference
  and re-uploading.
```